src/collision.cpp

```diff
@@ -527,6 +527,10 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 				is_collision = false;
 
 			CollisionInfo info;
+			if (is_collision) {
+				info.new_pos = *pos_f / BS;
```

Comment:
Division should be moved to `push_collision_move_result`. Please don't mix Minetest internals (factor `BS`) with exposed values.